src/raft.rs

```diff
@@ -1653,10 +1664,14 @@ impl<T: Storage> Raft<T> {
             }
         }
 
-        if send_append && m.from_delegate == INVALID_ID {
-            let from = m.from;
+        // TODO(qupeng): make it correct when `from`'s leader is itself.
+        if send_append && self.groups.get_delegate_by_member(m.from).is_none() {
```

Comment:
Why `self.groups.get_delegate_by_member(m.from).is_none()` ?

Reply:
If the `append response` comes from follower behind a delegate, we don't need to call `send_append` for it.

Reply:
The comment is wrong. Should be `from`'s delegate.

Reply:
Can we use `from_delegate` here?
Receiving a `reject` message with `from_delegate`, the leader knows it comes from a delegate.
And as a delegated follower don't send `reject` message to the leader, the `send_append` could be always false I think?

Reply:
Ok. I will use `from_delegate` here. BTW do you think we need to rename `from_delegate` to `delegate`?